packages/@react-spectrum/s2/src/Toast.tsx

```diff
@@ -0,0 +1,579 @@
+/*
+ * Copyright 2025 Adobe. All rights reserved.
+ * This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+
+import {ActionButton} from './ActionButton';
+import AlertIcon from '../s2wf-icons/S2_Icon_AlertTriangle_20_N.svg';
+import {Button} from './Button';
+import {CenterBaseline} from './CenterBaseline';
+import CheckmarkIcon from '../s2wf-icons/S2_Icon_CheckmarkCircle_20_N.svg';
+import Chevron from '../s2wf-icons/S2_Icon_ChevronDown_20_N.svg';
+import {CloseButton} from './CloseButton';
+import {createContext, ReactNode, useContext, useEffect, useMemo, useRef} from 'react';
+import {DOMProps} from '@react-types/shared';
+import {filterDOMProps, useEvent} from '@react-aria/utils';
+import {flushSync} from 'react-dom';
+import {focusRing, style} from '../style' with {type: 'macro'};
+import {FocusScope, useModalOverlay} from 'react-aria';
+import InfoIcon from '../s2wf-icons/S2_Icon_InfoCircle_20_N.svg';
+// @ts-ignore
+import intlMessages from '../intl/*.json';
+import {mergeStyles} from '../style/runtime';
+import {ToastOptions as RACToastOptions, UNSTABLE_Toast as Toast, UNSTABLE_ToastContent as ToastContent, UNSTABLE_ToastList as ToastList, ToastProps, UNSTABLE_ToastQueue as ToastQueue, UNSTABLE_ToastRegion as ToastRegion, ToastRegionProps, UNSTABLE_ToastStateContext as ToastStateContext} from 'react-aria-components';
+import {Text} from './Content';
+import toastCss from './Toast.module.css';
+import {useLocalizedStringFormatter} from '@react-aria/i18n';
+import {useMediaQuery} from '@react-spectrum/utils';
+import {useOverlayTriggerState} from 'react-stately';
+
+export type ToastPlacement = 'top' | 'top end' | 'bottom' | 'bottom end';
+export interface ToastContainerProps extends Omit<ToastRegionProps<SpectrumToastValue>, 'queue' | 'children'> {
+  /**
+   * Placement of the toast container on the page.
+   * @default "bottom"
+   */
+  placement?: ToastPlacement
+}
+
+export interface ToastOptions extends Omit<RACToastOptions, 'priority'>, DOMProps {
+  /** A label for the action button within the toast. */
+  actionLabel?: string,
+  /** Handler that is called when the action button is pressed. */
+  onAction?: () => void,
+  /** Whether the toast should automatically close when an action is performed. */
+  shouldCloseOnAction?: boolean
+}
```

Comment:
Would be nice to mention that the minimum time that the toast stays open is 5 seconds in the prop description, otherwise people might get confused why their provided timeout doesn't seem to work. Would also be nice to tie the storybook onAction to a control but that can be followup